res/skins/QMLDemo/EffectUnit.qml

```diff
@@ -53,7 +57,12 @@ Item {
                     anchors.right: effectMetaKnob.left
                     anchors.margins: 5
                     // TODO: Add a way to retrieve effect names here
-                    model: ["---", "Effect 1", "Effect 2", "Effect 3", "Effect 4"]
+                    textRole: "display"
+                    model: Mixxx.EffectsManager.visibleEffectsModel
+                    onActivated: {
+                        console.warn(1, root.unitNumber, effect.effectNumber, model.get(index).effectId);
+                        Mixxx.EffectsManager.loadEffect(1, root.unitNumber, effect.effectNumber, model.get(index).effectId);
```

Comment:
I'm not sure I like this... Though I don't really have anything better to propose...

Reply:
The question is how do we receive changes. I checked the controls and didn't see any COs that indicate the currently loaded effect. Therefore I think we need to add a QmlEffectChainSlotProxy that can emit signals when the effect is changed. If we do that, we can move the load method there.

Reply:
Binding to (`EffectRack1_EffectUnitN_EffectM`,`loaded`) might work. It should emit a "trigger" signal even if the value didn't change. But in the long term, we do need `QmlEffectChainSlotProxy` anyways so we can configure the parameters and so forth. We might even consider trying to remove the GUI changes from #2618 and redo that part based on QML.

Reply:
I don't think we actually need it for parameter configuration except the names.

Reply:
Well we don't really need it, but I find the current way its implemented with COs very hacky and would like to get rid of it in the longterm actually..